### PR TITLE
Azure OpenAI Service Connector

### PR DESCRIPTION
**Pull request by Laskewitz**
Added Azure OpenAI Service Connector as a custom connector.

image
image

### CODE REVIEW

File: custom-connectors/AzureOpenAIService/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,657 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure OpenAI Service",
+    "description": "Azure OpenAI Service provides REST API access to OpenAI's powerful language models including the GPT-3, Codex and Embeddings model series. These models can be easily adapted to your specific task including but not limited to content generation, summarization, semantic search, and natural language to code translation.",
```
Comment (Amjed-Ayoub) [inline custom-connectors/AzureOpenAIService/apiDefinition.swagger.json:5]: @Laskewitz is it possible to remove the word "API" from the description. Please see [here ](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector)and [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#connector-info).

File: custom-connectors/AzureOpenAIService/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/openai/deployments/{deployment-id}/completions": {
+      "post": {
+        "responses": {
+          "200": {
+            "description": "default",
```
Comment (Amjed-Ayoub) [inline custom-connectors/AzureOpenAIService/apiDefinition.swagger.json:43]: @Laskewitz is it possible to use more descriptive here. you can use something like "OK", "Success", "200", etc.
Please fix all responses descriptions in the swagger file.

File: custom-connectors/AzureOpenAIService/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+      },
+      "delete": {
+        "summary": "Delete deployment",
+        "description": "Deletes the specified deployment.",
+        "operationId": "DeleteDeployment",
+        "x-ms-visibility": "advanced",
+        "responses": {
+          "200": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline custom-connectors/AzureOpenAIService/apiDefinition.swagger.json:375]: @Laskewitz empty schema is not allowed, can you please define a proper schema for the response.
Please fix all empty schemas in the swagger file

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: custom-connectors/AzureOpenAIService/apiDefinition.swagger.json (hunk 2 above)
Comment (Laskewitz) [reply custom-connectors/AzureOpenAIService/apiDefinition.swagger.json]: Done and fixed.

File: custom-connectors/AzureOpenAIService/apiDefinition.swagger.json (hunk 1 above)
Comment (Laskewitz) [reply custom-connectors/AzureOpenAIService/apiDefinition.swagger.json]: Done and fixed.

File: custom-connectors/AzureOpenAIService/apiDefinition.swagger.json (hunk 3 above)
Comment (Amjed-Ayoub) [reply custom-connectors/AzureOpenAIService/apiDefinition.swagger.json:375]: Okay we will accept this as a custom connector. please be aware that this is not allowed for certified connectors.
Thank you so much again.

**Review by Amjed-Ayoub — APPROVED** [review]

**Pull request merged**